command-expansion-server/src/app.ts

```diff
@@ -83,8 +82,35 @@ app.post('/put-expansion', async (req, res) => {
   return;
 });
 
-app.put('/expansion-set', async (req, res) => {
-  res.status(501).send('PUT /expansion-set: Not implemented');
+app.post("/put-expansion-set", async (req, res) => {
+  const commandDictionaryId = req.body.input.commandDictionaryId as number;
+  const missionModelId = req.body.input.missionModelId as number;
+  const expansionIds = req.body.input.expansionIds as number[];
+
+  const { rows } = await db.query(`
+    WITH expansion_set_id AS (
+      INSERT INTO expansion_set (command_dict_id, mission_model_id)
+        VALUES ($1, $2)
+        RETURNING id
+    )
+    INSERT INTO expansion_set_to_rule (set_id, rule_id)
+      SELECT * FROM unnest(
+        array_fill((SELECT id FROM expansion_set_id), ARRAY[array_length($3::int[], 1)]),
+        $3::int[]
+      )
+    RETURNING (SELECT id FROM expansion_set_id);
+  `, [
+    commandDictionaryId,
+    missionModelId,
+    expansionIds,
+  ]);
+
+  if (rows.length < 1) {
+    throw new Error(`PUT /expansion-set: No expansion set was inserted in the database`);
+  }
+  const id = rows[0].id;
+  console.log(`PUT /expansion-set: Updated expansion set in the database: id=${id}`);
```

Comment:
For correctness I'd change this to read:
```
POST /put-expansion-set
```
Same suggestion for a few lines above in `throw new Error(...)`.